example/lib/use_restorable_property.dart

```diff
@@ -0,0 +1,46 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+
+/// This example demonstrate what is required for and how to use
+/// [useRestorationProperty].
+class RestorablePropertyExample extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('useRestorationProperty example'),
+      ),
+      // Any app using the restoration system needs to make the root
+      // `RestorationBucket` available to the widget tree. Usually this widget
+      // is placed high up in the tree, for example above `MaterialApp`.
+      body: RootRestorationScope(
+        restorationId: 'root',
+        // This widget provides hooks access to a `RestorationBucket`. It is a
+        // direct replacement for `RestorationScope` from the framework.
+        child: HookRestorationScope(
```

Comment:
We shouldn't need such widget. Relying on an InheritedWidget to work around the `RestorationMixin` could be a significant source of problems.

I would expect people to easily forget to wrap a custom `HookWidget` using `useRestorableProperty` into a `HookRestorationScope` – and in fact, they shouldn't have to. 

`useRestorableProperty` should be able to handle everything by itself

Reply:
I agree, it's not optimal. The reason I implemented it this way is that the implementation of `RestorableProperty` is tied to `RestorationMixin`. I have not looked into if it would be possible to use `RestorableProperty` without `RestorationMixin` but that doesn't seem like a good idea. If users can't use `RestorableProperty`, they also can't use all the subclasses which would mean quite a large part of the user facing restoration API would need to be replaced somehow. I'll take a second look at how to get around `HookRestorationScope`.

Reply:
The problem is, we can't reasonably expose the `RestorationMixin` through an InheritedWidget

That could cause major issues, as the mixin could suddenly be associated with multiple `HookWidget`